infrastructure/keymanager/src/mnemonic.rs

```diff
@@ -0,0 +1,460 @@
+            Err(_e) => assert!(false),
+        }
+        match find_mnemonic_word_from_index(desired_index, &MnemonicLanguage::Korean) {
+            Ok(word) => assert_eq!(desired_word, word),
+            Err(_e) => assert!(false),
+        }
+
+        //Encoding and Decoding using Spanish Simplified
+        let desired_index = 345;
+        let desired_word = MNEMONIC_SPANISH_WORDS[desired_index].to_string();
+        match find_mnemonic_index_from_word(&desired_word, &MnemonicLanguage::Spanish) {
+            Ok(index) => assert_eq!(desired_index, index),
+            Err(_e) => assert!(false),
+        }
+        match find_mnemonic_word_from_index(desired_index, &MnemonicLanguage::Spanish) {
+            Ok(word) => assert_eq!(desired_word, word),
+            Err(_e) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_mnemonic_from_bytes_and_to_bytes() {
+        let mut rng = rand::OsRng::new().unwrap();
+        let secretkey_bytes = SecretKey::random(&mut rng).to_vec();
+        match mnemonic::from_bytes(secretkey_bytes.clone(), &MnemonicLanguage::English) {
+            Ok(mnemonic_seq) => match mnemonic::to_bytes(&mnemonic_seq) {
+                Ok(mnemonic_bytes) => {
+                    let mismatched_bytes =
+                        secretkey_bytes.iter().zip(mnemonic_bytes.iter()).filter(|&(a, b)| a != b).count();
+                    assert_eq!(mismatched_bytes, 0);
+                }
+                Err(_e) => assert!(false),
+            },
+            Err(_e) => assert!(false),
+        }
+    }
+
+    #[test]
+    fn test_secretkey_to_mnemonic_and_from_mnemonic() {
+        //Valid Mnemonic sequence
+        let mut rng = rand::OsRng::new().unwrap();
+        let desired_k = SecretKey::random(&mut rng);
+        match desired_k.to_mnemonic(&MnemonicLanguage::Japanese) {
+            Ok(mnemonic_seq) => {
+                //Language not known
+                match SecretKey::from_mnemonic(&mnemonic_seq) {
+                    Ok(mnemonic_k) => assert_eq!(desired_k, mnemonic_k),
+                    Err(_e) => assert!(false),
+                }
+                //Language known
+                match SecretKey::from_mnemonic_with_language(&mnemonic_seq, &MnemonicLanguage::Japanese) {
+                    Ok(mnemonic_k) => assert_eq!(desired_k, mnemonic_k),
+                    Err(_e) => assert!(false),
+                }
+            }
+            Err(_e) => assert!(false),
+        }
+
+        //Invalid Mnemonic sequence
+        let mnemonic_seq = vec![
+            "clever", "jaguar", "bus", "engage", "oil", "august", "media", "high", "trick", "remove", "tiny", "join",
+            "item", "tobacco", "orange", "pny", "tomorrow", "also", "dignity", "giraffe", "little", "board", "army",
+        ]
+        .iter()
+        .map(|x| x.to_string())
+        .collect::<Vec<String>>();
+        //Language not known
+        match SecretKey::from_mnemonic(&mnemonic_seq) {
```

Comment:
One liner 

```rust
assert!(SecretKey::from_mnemonic(&mnemonic_seq).is_err())
```
would do the same thing - same for below

Reply:
I can't replace this match with the one-liner as I need to check that the original key matches the decoded key: assert_eq!(desired_k, mnemonic_k). The match is used to unwrap the decoded key.